addons/beehave/nodes/beehave_node.gd

```diff
@@ -27,3 +28,14 @@ func tick(actor: Node, blackboard: Blackboard) -> int:
 ## Called when this node needs to be interrupted before it can return FAILURE or SUCCESS.
 func interrupt(actor: Node, blackboard: Blackboard) -> void:
 	pass
+
+
+## Called before the first time it ticks by the parent.
+func enter(actor: Node, blackboard: Blackboard) -> void:
+	pass
```

Comment:
I am wondering if there would ever be a performance impact of having these methods. Technically, `pass` will not execute anything so I'd expect this to be optimised at runtime to not take up any additional ms to compute.

Reply:
Would not this be an over optimization though?  The engine also have a lot of node methods that can but are not usually overriden. If the goal is to make it as fast as possible we should be writing in c++. If this slows down considerably we could take a look.

Reply:
Great point, I have not considered that!